delivery_package_number/wizard/stock_backorder_confirmation.py

```diff
@@ -15,9 +15,13 @@ class StockBackorderConfirmation(models.TransientModel):
     @api.depends("pick_ids")
     def _compute_ask_number_of_packages(self):
         for item in self:
-            item.ask_number_of_packages = bool(item.pick_ids.carrier_id)
+            item.ask_number_of_packages = bool(
+                item.pick_ids.carrier_id and not item.pick_ids._origin.package_ids
+            )
 
     def process(self):
+        ctx = self._context.copy()
         if self.number_of_packages:
+            ctx.update({"set_default_package": False})
             self.pick_ids.write({"number_of_packages": self.number_of_packages})
-        return super().process()
+        return super(StockBackorderConfirmation, self.with_context(**ctx)).process()
```

Comment:
No need of copying the full context, and put a comment why you are doing this:

```suggestion
        # put context key for avoiding `base_delivery_carrier_label` auto-packaging feature
        return super(StockBackorderConfirmation, self.with_context(set_default_package=False)).process()
```